src/components/space/functionality/ExportData.tsx

```diff
@@ -1,45 +1,161 @@
 
-import CloudDownloadIcon from '@mui/icons-material/CloudDownload';
-import DoneIcon from '@mui/icons-material/Done';
-import LoadingButton from '@mui/lab/LoadingButton';
-import { Tooltip } from '@mui/material';
+import ArrowDropDownIcon from '@mui/icons-material/ArrowDropDown';
+import {
+  Box,
+  Button,
+  ButtonGroup,
+  ClickAwayListener,
+  Grow,
+  List,
+  ListItem,
+  ListItemText,
+  MenuItem,
+  MenuList,
+  Paper,
+  Popper,
+  Typography,
+  styled,
+} from '@mui/material';
+
+import { ExportActionsFormatting } from '@graasp/sdk';
+
+import { Braces, Grid3X3 } from 'lucide-react';
 
 import { useAnalyticsTranslation } from '@/config/i18n';
 
 import { mutations } from '../../../config/queryClient';
 
+const CustomListItem = styled(ListItem)(({ theme }) => ({
+  paddingTop: 0,
+  '&:before': {
+    content: '"•"',
+    paddingRight: theme.spacing(1),
+    fontSize: theme.typography.body1.fontSize,
+  },
+}));
+
 const ExportData = (): JSX.Element => {
   const { t } = useAnalyticsTranslation();
-  const [clicked, setClicked] = useState(false);
-  const { mutate: exportActions, isLoading } = mutations.useExportActions();
+  const [format, setFormat] = useState(ExportActionsFormatting.CSV);
+  const [open, setOpen] = useState(false);
+  const anchorRef = useRef<HTMLDivElement>(null);
+
+  const [isFormatExported, setIsFormatExported] = useState({
+    [ExportActionsFormatting.CSV]: false,
+    [ExportActionsFormatting.JSON]: false,
+  });
+
+  const { mutate: exportActions } = mutations.useExportActions();
   const { itemId } = useParams();
 
+  const icons = {
+    [ExportActionsFormatting.CSV]: <Grid3X3 size={16} />,
+    [ExportActionsFormatting.JSON]: <Braces size={16} />,
+  };
   const onClick = () => {
     if (itemId) {
-      setClicked(true);
-      exportActions(itemId);
+      exportActions({ itemId, format });
+      setIsFormatExported({ ...isFormatExported, [format]: true });
+    }
+  };
+
+  const handleMenuItemClick = (option: ExportActionsFormatting) => {
+    setFormat(option);
+    setOpen(false);
+  };
+
+  const handleToggle = () => {
+    setOpen((prevOpen) => !prevOpen);
+  };
+
+  const handleClose = (event: Event) => {
+    if (
+      anchorRef.current &&
+      anchorRef.current.contains(event.target as HTMLElement)
+    ) {
+      return;
     }
+
+    setOpen(false);
   };
 
+  const formats = Object.values(ExportActionsFormatting).slice().reverse();
   return (
-    <Tooltip
-      title={clicked ? t('EXPORT_SUCCESS_MESSAGE') : t('EXPORT_TOOLTIP')}
-      placement="right"
-      arrow
-    >
-      <LoadingButton
-        onClick={onClick}
-        variant="contained"
-        endIcon={clicked ? <DoneIcon /> : <CloudDownloadIcon />}
-        disabled={clicked}
-        loading={isLoading}
-        size="large"
+    <>
+      <Typography variant="h6">{t('SELECT_FORMAT_DIALOG_TITLE')}</Typography>
+      <Typography variant="body1">
+        {t('SELECT_FORMAT_DIALOG_DESCRIPTION')}
```

Comment:
Do you think you could use `constants` as we do in the other projects? You don't have to move everything, but at least the new one you create.